salt/haproxy/config/haproxy.cfg.jinja

```diff
@@ -248,8 +252,7 @@ listen {{ name }}
     {{ line }}
     {% endfor %}
 
-    {{ "{{" }}range service "{{ config.service }}@{{ pillar.dc }}"}}
-    {% raw %}server {{.Node}} {{.Address}}:{{.Port}} check{{end}}{% endraw %}{% if config.get("send_proxy", False) %} send-proxy{% endif %}
+    server-template backend 1-3 _{{ config.service }}._tcp.service.{{ pillar.dc }}.consul resolvers consul resolve-opts allow-dup-ip resolve-prefer ipv4 check {% if config.get("send_proxy", False) %} send-proxy{% endif %}
```

Comment:
is there a way to only template out 3 backends if we need them?

Reply:
I suppose we could make the "count" tunable in service configuration. 3 was just "plucked from air", but since we only ever tend to use one backend, maybe 2 is a better default choice.

Reply:
<img width="210" alt="Screenshot 2024-08-29 at 10 17 12 AM" src="https://github.com/user-attachments/assets/b94d254c-7395-4713-9dc8-89ac874a3b1a">

Reply:
Implemented in 776fdd3, one drawback is that if we want multiple backends to be available during a migration we would need to ensure we update the "backends" value. Thoughts?

Reply:
I think thats an acceptable thing, we just need docs on it. It will for sure unmuddy the LB dashboards 😬

Reply:
but 2 is better than 3, so either way

Reply:
added some documentation in f15ab08